scrapy_autounit/middleware.py

```diff
@@ -1,136 +1,51 @@
-import os
-import six
 import pickle
-import random
 import logging
 
+from scrapy import signals
 from scrapy.exceptions import NotConfigured
-from scrapy.commands.genspider import sanitize_module_name
 
-from .utils import (
-    add_sample,
-    write_test,
-    response_to_dict,
-    get_or_create_test_dir,
-    parse_request,
-    get_project_dir,
-    get_middlewares,
-    create_dir,
-    parse_callback_result,
-    clear_fixtures,
-    get_filter_attrs,
-)
-
-logger = logging.getLogger(__name__)
+from .recorder import Recorder
 
 
-def _copy_settings(settings):
-    out = {}
-    for name in settings.getlist('AUTOUNIT_INCLUDED_SETTINGS', []):
-        out[name] = settings.get(name)
-    return out
+logger = logging.getLogger(__name__)
 
 
 class AutounitMiddleware:
     def __init__(self, crawler):
+        self.crawler = crawler
         settings = crawler.settings
-        spider = crawler.spider
 
-        if not any(
-            self.__class__.__name__ in s
-            for s in settings.getwithbase('SPIDER_MIDDLEWARES').keys()
-        ):
+        spider_mw = settings.getwithbase('SPIDER_MIDDLEWARES').keys()
+        if not any(self.__class__.__name__ in mw for mw in spider_mw):
             raise ValueError(
-                '%s must be in SPIDER_MIDDLEWARES' % (
-                    self.__class__.__name__,))
+                '{} must be in SPIDER_MIDDLEWARES'.format(
+                    self.__class__.__name__))
```

Comment:
This checks that you don't place `scrapy_autounit.AutounitMiddleware` in `DOWNLOADER_MIDDLEWARES` instead of `SPIDER_MIDDLEWARES`. Happened once in a project so we added this check.
Maybe it can be written more explicitly though. Since it doesn't cover that you don't place the middleware in both lists, for example.